static/downloadables/parse_associations.py

```diff
@@ -487,36 +460,89 @@ def parse_vcf(filteredFilePath, clumpsObjDict, tableObjDict, snpSet, clumpNumDic
+                                                index_pvalue = tableObjDict['associations'][index_snp]['traits'][trait][study][pValBetaAnnoValType]['pValue']
 
                                                 # Check whether the existing index snp or current snp have a lower pvalue for this study
                                                 # and switch out the data accordingly
                                                 if pValue < index_pvalue:
-                                                    index_snp_map[sample][clumpNum] = rsID, alleles
+                                                    index_snp_map[sample][clumpNum] = rsID, alleles if complements is None else complements
                                                     clumpedVariants.add(index_snp)
                                                 else:
                                                     clumpedVariants.add(rsID)
                                             else:
                                                 # Since the clump number for this snp position and study/name
                                                 # doesn't already exist, add it to the index map and the sample map
-                                                index_snp_map[sample][clumpNum] = rsID, alleles
+                                                index_snp_map[sample][clumpNum] = rsID, alleles if complements is None else complements
                                         else:
                                             # Since the study/name combo wasn't already used in the index map, add it to both the index and sample map
-                                            index_snp_map[sample][clumpNum] = rsID, alleles
+                                            index_snp_map[sample][clumpNum] = rsID, alleles if complements is None else complements
 
+                            #TODO: LOOK AT THIS
                             # the sample's alleles don't include the risk allele and early clumping is not requested
                             else:
-                                if sample not in viable_snp_counter:
-                                    no_viable_snp_counter.add(sample)
                                 unmatchedAlleleVariants.add(rsID)
 
                             clumped_snps_map[sample] = clumpedVariants
                             neutral_snps_map[sample] = unmatchedAlleleVariants
 
-        # Check to see which study/sample combos didn't have any viable snps
-        # and create blank entries for the sample map for those that didn't
-        for sample in no_viable_snp_counter:
-            sample_map[sample][""] = ""
+        # This next code accounts for snps that are in the study but are not reported in the sample. Instead of assuming the reference allele, we 
+        # assume that the allele is unknown and thus will use MAF for calculations of these snps
+        snpsLeftToImpute = set(snpSet).difference(usedSnps)
```

Comment:
Is this duplicate code from above? If so, can it be put in a separate function, especially since it's so large and could "mutate?"

Reply:
It is very similar, but it is not an exact duplication, since with one we have to worry about it for only one sample and the other we have to handle it for all the samples